litrpc/lit-accounts.proto

```diff
@@ -0,0 +1,122 @@
+syntax = "proto3";
+
+package litrpc;
+
+option go_package = "github.com/lightninglabs/lightning-terminal/litrpc";
+
+service Accounts {
+    /*
+    CreateAccount adds an entry to the account database. This entry represents
+    an amount of satoshis (account balance) that can be spent using off-chain
+    transactions (e.g. paying invoices).
+
+    Macaroons can be created to be locked to an account. This makes sure that
+    the bearer of the macaroon can only spend at most that amount of satoshis
+    through the daemon that has issued the macaroon.
+
+    Accounts only assert a maximum amount spendable. Having a certain account
+    balance does not guarantee that the node has the channel liquidity to
+    actually spend that amount.
+    */
+    rpc CreateAccount (CreateAccountRequest) returns (CreateAccountResponse);
+
+    /*
+    UpdateAccount updates an existing account in the account database.
+    */
+    rpc UpdateAccount (UpdateAccountRequest) returns (Account);
+
+    /*
+    ListAccounts returns all accounts that are currently stored in the account
+    database.
+    */
+    rpc ListAccounts (ListAccountsRequest) returns (ListAccountsResponse);
+
+    /*
+    RemoveAccount removes the given account from the account database.
+    */
+    rpc RemoveAccount (RemoveAccountRequest) returns (RemoveAccountResponse);
+}
+
+message CreateAccountRequest {
+    /*
+    The initial account balance in satoshis representing the maximum amount that
+    can be spent.
+    */
+    uint64 account_balance = 1;
+
+    /*
+    The expiration date of the account as a timestamp. Set to 0 to never expire.
+    */
+    int64 expiration_date = 2;
+}
+
+message CreateAccountResponse {
+    // The new account that was created.
+    Account account = 1;
+
+    // The macaroon with all permissions required to access the account.
+    bytes macaroon = 2;
+}
+
+message Account {
+    // The ID of the account.
+    string id = 1;
+
+    /*
+    The initial balance in satoshis that was set when the account was created.
+    */
+    uint64 initial_balance = 2;
+
+    // The current balance in satoshis.
+    int64 current_balance = 3;
+
+    // Timestamp of the last time the account was updated.
+    int64 last_update = 4;
+
+    /*
+    Timestamp of the account's expiration date. Zero means it does not expire.
+    */
+    int64 expiration_date = 5;
+
+    /*
+    The list of invoices created by the account. An invoice created by an
+    account will credit the account balance if it is settled.
+    */
+    repeated bytes invoices = 6;
+
+    /*
+    The list of payments made by the account. A made by an account will debit
+    the account balance if it is settled.
+    */
+    repeated bytes payments = 7;
```

Comment:
also: for both payments and invoices about, could be nice to show the amounts associated with them and also show which have completed vs are still pending.

Reply:
Yeah, that might be nice. Though we don't have a state or amount for the invoice currently. Maybe a nice optimization for the future. I changed both response types to their own message types so we can easily extend this in the future.

Reply:
awesome